src/components/dialog-parameters/parameters.js

```diff
@@ -0,0 +1,220 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { FormattedMessage } from 'react-intl';
+
+import { useSelector } from 'react-redux';
+
+import makeStyles from '@mui/styles/makeStyles';
+import {
+    Grid,
+    Box,
+    Button,
+    Container,
+    Dialog,
+    DialogContent,
+    DialogTitle,
+    Switch,
+    Tab,
+    Tabs,
+    Typography,
+} from '@mui/material';
+import { useSnackbar } from 'notistack';
+
+import { updateConfigParameter } from '../../utils/rest-api';
+import {
+    displayErrorMessageWithSnackbar,
+    useIntlRef,
+} from '../../utils/messages';
+import { SingleLineDiagramParameters } from './single-line-diagram-parameter';
+import { MapParameters } from './map-parameter';
+import { LoadFlow } from './load-flow-parameter';
+import { NetworkParameters } from './network-parameter';
+
+export const useStyles = makeStyles((theme) => ({
+    title: {
+        padding: theme.spacing(2),
+    },
+    grid: {
+        padding: theme.spacing(2),
+    },
+    minWidthMedium: {
+        minWidth: theme.spacing(20),
+    },
+    controlItem: {
+        justifyContent: 'flex-end',
+    },
+    button: {
+        marginBottom: '30px',
+    },
+    advancedParameterButton: {
+        marginTop: theme.spacing(3),
+        marginBottom: theme.spacing(1),
+    },
+}));
+
+export function useParameterState(paramName) {
+    const intlRef = useIntlRef();
+
+    const { enqueueSnackbar } = useSnackbar();
+
+    const paramGlobalState = useSelector((state) => state[paramName]);
+
+    const [paramLocalState, setParamLocalState] = useState(paramGlobalState);
+
+    useEffect(() => {
+        setParamLocalState(paramGlobalState);
+    }, [paramGlobalState]);
+
+    const handleChangeParamLocalState = useCallback(
+        (value) => {
+            setParamLocalState(value);
+            updateConfigParameter(paramName, value).catch((errorMessage) => {
+                setParamLocalState(paramGlobalState);
+                displayErrorMessageWithSnackbar({
+                    errorMessage: errorMessage,
+                    enqueueSnackbar: enqueueSnackbar,
+                    headerMessage: {
+                        headerMessageId: 'paramsChangingError',
+                        intlRef: intlRef,
+                    },
+                });
+            });
+        },
+        [
+            paramName,
+            enqueueSnackbar,
+            intlRef,
+            setParamLocalState,
+            paramGlobalState,
+        ]
+    );
+
+    return [paramLocalState, handleChangeParamLocalState];
+}
+
+export function MakeSwitch(prop, label, callback) {
```

Comment:
this should be a component 
const SwitchWithLabel = ({value, label, onChange})
and the same for each ones